app/controllers/users/confirmations_controller.rb

```diff
@@ -112,7 +112,12 @@ def after_confirmation_path_for(resource)
     end
 
     def process_confirmed_user
-      analytics.track_event('Email changed and confirmed', @confirmable)
+      if @confirmable.previous_changes.key?('email')
+        analytics.track_event('Email change confirmed', @confirmable)
+        create_user_event(:email_changed, @confirmable)
+      else
+        analytics.track_event('Email address confirmed', @confirmable)
+      end
```

Comment:
The other advantage of putting the call in the Analytics class is it enables this functionality by simply adding one line to the Analytics class, without needing to modify the existing controller code.